service/frontend/namespace_handler.go

```diff
@@ -687,6 +687,14 @@ func (d *namespaceHandler) CreateWorkflowRule(
 
 	if config.WorkflowRules == nil {
 		config.WorkflowRules = make(map[string]*rulespb.WorkflowRule)
+	} else {
+		maxRules := d.config.MaxWorkflowRulesPerNamespace(nsName)
+		if len(config.WorkflowRules) > maxRules {
```

Comment:
nit: `>=` (here and below). Otherwise we'll have upto max+1 rules per namespace.